worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -1478,29 +1488,45 @@ public int makeSphere(Vector pos, Pattern block, double radiusX, double radiusY,
                         }
                     }
 
-                    if (setBlock(pos.add(x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, -z), block)) {
-                        ++affected;
+                    if (directionVector != null) {
+                        Vector vectors[] = {pos.add(x, y, z), pos.add(-x, y, z), pos.add(x, y, -z), pos.add(-x, y, -z), pos.add(x, -y, z), pos.add(-x, -y, z), pos.add(x, -y, -z), pos.add(-x, -y, -z)};
```

Comment:
I didn't really notice this, but why is there an "if (directionVector != null)" and still genUp and genDown?
You can implement the later using directionVector as well.
It'll be less code, too.
Btw, having everything in an array might not be that bad if that allows you to put all the logic in one piece of code instead of 6.